src/vsr/client_replies.zig

```diff
@@ -253,8 +264,13 @@ pub fn ClientRepliesType(comptime Storage: type) type {
             }
         }
 
+        pub fn remove_reply(client_replies: *ClientReplies, slot: Slot) void {
+            client_replies.faulty.unset(slot.index);
+        }
+
         pub fn write_reply(client_replies: *ClientReplies, slot: Slot, message: *Message) void {
             assert(client_replies.ready_callback == null);
+            assert(client_replies.writes.available() > 0);
```

Comment:
Hm, why woundn't `create_client_table_entry` trip this assert?

Reply:
I forgot to add a comment, but:

```
        /// The caller is responsible for ensuring that the ClientReplies is able to write
        /// by calling `write_reply()` after `ready()` finishes.
```

Replica inverted this order, calling `ready` _after_ instead of _before_. This was correct but confusing, so I am moving it to before.

Reply:
Ah, I guess the problem is that this maybe is wrong then?

https://github.com/tigerbeetledb/tigerbeetle/blob/73ddb9b9acc8c5db76ab729393f87626b0e1a842/src/vsr/replica.zig#L2800

Reply:
Ah, that can be
```
            assert(self.client_replies.writes.available() > 0);
```
now